db/db_impl.cc

```diff
@@ -5106,6 +5108,14 @@ void DBImpl::MaybeFlushColumnFamilies() {
 
   unable_to_flush_oldest_log_ = false;
 
+  // Another thread could have removed some logs from alive_log_files_ while
+  // we were in this method. This is possible because SwitchMemtable() unlocks
+  // and re-locks mutex_.
+  if (alive_log_files_.begin()->number != oldest_alive_log) {
+    assert(alive_log_files_.begin()->number > oldest_alive_log);
+    return;
+  }
+
```

Comment:
I think in 2pc we should do it to avoid unknown issues. For 2pc, I'm not sure whether it is still going to right. @reidHoruff  what do you think?